WooCommerce/Classes/ViewRelated/Products/Media/Media+WPMediaAsset.swift

```diff
@@ -0,0 +1,60 @@
+import MobileCoreServices
+import WPMediaPicker
+import Yosemite
+
+extension Media: WPMediaAsset {
+    public func image(with size: CGSize, completionHandler: @escaping WPMediaImageBlock) -> WPMediaRequestID {
+        let imageURL = thumbnailURL ?? src
+        guard let url = URL(string: imageURL) else {
+            return 0
+        }
+
+        let imageService = ServiceLocator.imageService
```

Comment:
> Moved the image data provider (where it fetches the image data given for a media item) from WPMediaAsset protocol to somewhere else like WPMediaCollectionDataSource. This way, we could DI any dependencies (like ImageService) to the data source, instead of having to use the app singleton ServiceLocator.imageService to fetch the image in the Media+WPMediaAsset extension.

Yup. This `ServiceLocator` access inside a model is definitely awkward. 😄 We definitely should fix this. Not high priority though. 

Perhaps we can add a comment here that we are planning to change this? I don't want to encourage future developers that this is "normal". 

Considering that the networking is already handled by WC, and the paging is suboptimal, as you said, I also would not be opposed to implementing our own. It's probably a solution for us to move faster. We can think about rolling our own when we have another "custom picker". 😱 Or maybe if we have to implement empty states? 😱 

In WPAndroid, the [MediaPicker library](https://github.com/wordpress-mobile/MediaPicker-Android) was even abandoned. I wonder if the reasons are similar. 🤔

Reply:
yup added TODO marks here: https://github.com/woocommerce/woocommerce-ios/pull/2060/commits/d45607a64270950b2b8ab29e34ba5c1f2c152383

I think it'd be great to improve our shared `WPMediaPicker` library by making it possible to extend any future functionality that we'd like. That said, the library was written in Objective-C so I would be down to write our own and make it a shared library as well 😄